crates/analyzer-host/src/tests/unit_tests.rs

```diff
@@ -0,0 +1,62 @@
+mod fsm_tests {
```

Comment:
Feel free to leave this in, but note that the file itself is already a module (called `unit_tests`), so there's no need to nest the tests in another.